src/viadot/orchestration/prefect/flows/transform_and_catalog.py

```diff
@@ -251,4 +280,14 @@ def transform_and_catalog(  # noqa: PLR0913
     )
     remove_dbt_repo_dir(dbt_repo_name, wait_for=wait_for)
 
+    if task_failed:
+        return Failed()
+
+    if not fail_flow_on_test_failure:
+        model_error_pattern = re.compile(
+            r"ERROR creating sql table model", re.IGNORECASE
+        )
+        if any(model_error_pattern.search(line) for line in build):
+            return Failed(message="One or more models failed to build.")
```

Comment:
Shouldn't it be the opposite?

```suggestion
    if fail_flow_on_test_failure:
        model_error_pattern = re.compile(
            r"ERROR creating sql table model", re.IGNORECASE
        )
        if any(model_error_pattern.search(line) for line in build):
            return Failed(message="One or more models failed to build.")
```

Reply:
Makes sense, but I have one more confusion here: the string you're looking for looks like a model materialization failure, and not a test failure. You even call it "model error pattern", not "test failure pattern".

Reply:
When `fail_flow_on_test_failure` is False, it means that user does not want to get failed flow in case of JUST test failures. In every case, when a model fails to build, the flow must finish in state Failed().

So this part is checking if a model failure was present. If it was, flow must finish in state Failed(). If it wasn't, it implies that all models refreshed properly, and tests might have or might have not failed. When `fail_flow_on_test_failure` is set to False, user does not care if the tests failed.

Reply:
Yeah so this is about model build failure, not test failure, right? Model building can also fail if the model itself is incorrect (eg. due to error in SQL code), even before any tests are executed. So maybe rename things to reflect that this is about model build failures, not about model test failures specifically?